distributed/scheduler.py

```diff
@@ -2366,6 +2366,7 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
         if (
             valid_workers is None
             and len(group) > self._total_nthreads * 2
+            and len(group._dependencies) < 5
             and sum(map(len, group._dependencies)) < 5
```

Comment:
I'm surprised that this is relevant.  I would expect almost all groups to have less than five group dependencies.